.github/workflows/test.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and Test
+
+on: push
+
+jobs:
+  manager:
+    name: Build and Test
+    
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # TODO run on Windows as well
+        os: [ubuntu-latest, macos-latest]
```

Comment:
@fortuna @mpmcroy the '-latest' targets change as new versions come out.  I was debating listing specific ubuntu and macos versions here just to increase our coverage.  On the other hand, this would significantly increase our rate of CI minute spending, especially running separate Mac and Windows versions, since

> Jobs that run on Windows and macOS runners that GitHub hosts consume minutes at 2 and 10 times the rate that jobs on Linux runners consume. For example, using 1,000 Windows minutes would consume 2,000 of the minutes included in your account. Using 1,000 macOS minutes, would consume 10,000 minutes included in your account. (https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions)

I settled that it would be best to keep to latest versions until we see a reason to need to test on multiple versions of a platform.  WDYT?

Reply:
I think running on `-latest` makes sense for pushes and PRs. It seems likely to catch most issues.

But maybe it would smart to run a full matrix of supported OSes and versions nightly on the master branch.

Reply:
We don't push to master enough to need nightly builds IMO.  I'd like to put this conversation in a separate PR, though, so that we can get testing working at a reasonable rate first and then work on improving our CI from there

Reply:
One thing to note: the matrix is really only telling the development platform. Not the client platform. The most important is Linux, since we use it for releases in the CI. Most of us already run macOS tests as part of the development these days, so that will ensure the tests are passing, or we will capture soon. 

My suggestion:
- Only run Linux on push. Fix the linux version to be whatever we use in the CI.
- Run macOS daily, or drop it if needed.
- If we decide to support Windows development at some point, we should add it to the daily run.